src/analytics/api/utils/pollutants/report.py

```diff
@@ -0,0 +1,140 @@
+import json
+from datetime import datetime
+import requests
+import pandas as pd
+from google.cloud import bigquery
+from google.oauth2 import service_account
+from config import Config
+
+from timezonefinder import TimezoneFinder
+import pytz
+
+def fetch_air_quality_data(grid_id, start_time, end_time) -> list:
+    # Convert start_time and end_time to ISO format
+    start_time_iso = start_time.isoformat() + 'Z'
+    end_time_iso = end_time.isoformat() + 'Z'
+
+    grid_params = {
+        "token": Config.AIRQO_API_TOKEN,
+        "startTime": start_time_iso,
+        "endTime": end_time_iso,
+        "recent": "no"
+    }
+
+    grid_url = f"https://platform.airqo.net/api/v2/devices/measurements/grids/{grid_id}"
+
+    try:
+        grid_response = requests.get(grid_url, params=grid_params)
+        grid_response.raise_for_status()  # Raise an exception for 4xx and 5xx status codes
+
+        data = grid_response.json()
+
+        # Extracting only the 'site_id' from each measurement
```

Comment:
Hi @wabinyai , if we are only interested in the site_id from the Grids measurement API...wont it be more optimal to just then use the Grids summary endpoint (`{{baseUrl}}/api/v2/devices/grids/summary`) which just returns site IDs and other small details associated with the Grid.

 Ideally and generally, most endpoint for measurements take more time than others which just return an entity's details.

Additionally, this Grids Summary endpoint is NOT protected so you can just call it WITHOUT a token. :)

Reply:
It will be resolved. On my end summary requests for token